src/services/api.ts

```diff
@@ -61,47 +59,13 @@ export enum Errors {
 	InvalidTokenId = "Invalid token id",
 }
 export default class Contract {
-	api: ApiInterface;
-	tokenCache: Map<TokenId, Token> = new Map<TokenId, Token>();
-	tokenList: TokenId[] = new Array<TokenId>();
-	lastUpdated = 0;
-
-	constructor() {
-		let mocked = CONTRACT_MOCKED;
-		if (typeof window !== "undefined" && window.localStorage) {
-			// Get info from local storage
-			const mockedValue = window.localStorage.getItem("mocked");
-			if (mockedValue !== null) {
-				mocked = mockedValue === "true";
-			}
-		}
-
-		this.api = mocked ? new ContractMock() : new GGxNetwork();
-	}
-
-	changeContract() {
-		if (Contract.isMocked()) {
-			this.api = new ContractMock();
-		} else {
-			this.api = new ContractMock();
-		}
-		this.tokenCache = new Map<TokenId, Token>();
-		this.tokenList = new Array<TokenId>();
-		this.lastUpdated = 0;
-	}
-
-	static isMocked(): boolean {
-		if (typeof window !== "undefined" && window.localStorage) {
-			const mockedValue = window.localStorage.getItem("mocked");
-			return mockedValue === "true";
-		}
-		return CONTRACT_MOCKED;
-	}
+	private api: ApiInterface;
+	private tokenCache: Map<TokenId, Token> = new Map<TokenId, Token>();
+	private tokenList: TokenId[] = new Array<TokenId>();
+	private lastUpdated = 0;
 
-	static setMocked(value: boolean) {
-		if (typeof window !== "undefined" && window.localStorage) {
-			window.localStorage.setItem("mocked", `${value}`);
-		}
+	constructor(api: ApiInterface) {
```

Comment:
How can I generate this api argument in type of ApiInterface?

Reply:
ApiInterface is an `interface` - `GgxNetwork` extends it.